test/int/msDebuggerPropertyId.test.ts

```diff
@@ -0,0 +1,185 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { createServer } from 'http-server';
+
+import * as ts from 'vscode-chrome-debug-core-testsupport';
+
+import * as testSetup from './testSetup';
+import * as assert from 'assert';
+import { DebugProtocol } from '../../node_modules/vscode-debugadapter/node_modules/vscode-debugprotocol/lib/debugProtocol';
+
+suite('msDebuggerPropertyId', () => {
+    const DATA_ROOT = testSetup.DATA_ROOT;
+
+    let dc: ts.ExtendedDebugClient;
+    setup(() => {
+        return testSetup.setup()
+            .then(_dc => dc = _dc);
+    });
+
+    let server: any;
+    teardown(() => {
+        if (server) {
+            server.close();
+        }
+
+        return testSetup.teardown();
+    });
+
+    suite('Test setting variable values:', () => {
+        test('Test updating local scope variable using Debugger.setVariableValue.', async () => {
+            const testProjectRoot = path.join(DATA_ROOT, 'msDebuggerPropertyId');
+            const scriptPath = path.join(testProjectRoot, 'src/script.ts');
+
+            server = createServer({ root: testProjectRoot });
+            server.listen(7890);
+
+            const url = 'http://localhost:7890/index.html';
+
+            const bpLine = 9;
+            const bpCol = 5;
+            const bpLine2 = 11;
+            const bpCol2 = 1;
+
+            // Hit the first breakpoint
+            await dc.hitBreakpointUnverified({ url, webRoot: testProjectRoot }, { path: scriptPath, line: bpLine, column: bpCol });
+
+            // Retrieve the variableReference for the Local scope
+            let st = await dc.stackTraceRequest();
+            let scopes = await dc.scopesRequest({ frameId: st.body.stackFrames[0].id });
+            let localsRef = scopes.body.scopes[0].variablesReference;
+            let variables = await dc.variablesRequest({ variablesReference: localsRef });
+
+            // Update the x variable on the local scope
+            await dc.setVariableRequest({ variablesReference: localsRef, name: 'x', value: '999' });
+
+            // Continue to end of function
+            await dc.setBreakpointsRequest({ source: { path: scriptPath }, breakpoints: [{ line: bpLine2, column: bpCol2 }] });
+            await dc.continueTo('breakpoint', { line: bpLine2, column: bpCol2 });
+            st = await dc.stackTraceRequest();
+            scopes = await dc.scopesRequest({ frameId: st.body.stackFrames[0].id });
+            localsRef = scopes.body.scopes[0].variablesReference;
+            variables = await dc.variablesRequest({ variablesReference: localsRef });
+
+            // Ensure x is updated
+            for (const variable of variables.body.variables) {
+                if (variable.name === 'x') {
+                    assert.equal(variable.value, '999');
```

Comment:
What do you think of adding a console.log('x is' + x) in the webpage so we can test that the program also sees x with the new value?

Reply:
Good call, I'm working on the update now

Reply:
I realized I meant to do this with the test the way it was written. I've added checks for the values of additional variables at the end of the function that are assigned the values of the variables we touch. I.e. y is assigned the value of x between break 1 and 2, and I confirm y=== updated x value